### PR TITLE
Configurable epoch change delay

### PR DESCRIPTION
**Pull request by miiu96**
## Reasoning behind the pull request
- Moved the hard-coded wait duration from the epoch start trigger in `config.toml` file

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: epochStart/shardchain/trigger.go [hunk 1]
```diff
@@ -260,6 +263,7 @@ func NewEpochStartTrigger(args *ArgsShardEpochStartTrigger) (*trigger, error) {
 		appStatusHandler:              args.AppStatusHandler,
 		roundHandler:                  args.RoundHandler,
 		enableEpochsHandler:           args.EnableEpochsHandler,
+		extraDelayForRequestBlockInfo: args.ExtraDelayForRequestBlockInfo,
```
Comment (iulianpascalau) [inline epochStart/shardchain/trigger.go:266]: I think we should add a warning in the logs if we provide a different value of 3 seconds for this (as seen in the chain simulator)
Reason: If we change this in the future, at least we should have a warning that the epoch change trigger might not behave normally.

File: epochStart/shardchain/trigger.go [hunk 2]
```diff
@@ -586,7 +590,7 @@ func (t *trigger) receivedMetaBlock(headerHandler data.HeaderHandler, metaBlockH
 		t.newEpochHdrReceived = true
 		t.mapEpochStartHdrs[string(metaBlockHash)] = metaHdr
 		// waiting for late broadcast of mini blocks and transactions to be done and received
-		wait := common.ExtraDelayForRequestBlockInfo
```
Comment (iulianpascalau) [inline epochStart/shardchain/trigger.go:589]: remove the constant?

**Review by iulianpascalau — COMMENTED** [review]
:+1:

**Review by sstanculeanu — COMMENTED** [review]
no other comments

File: epochStart/shardchain/trigger.go (hunk 1 above)
Comment (miiu96) [reply epochStart/shardchain/trigger.go:270]: done

File: epochStart/shardchain/trigger.go (hunk 2 above)
Comment (miiu96) [reply epochStart/shardchain/trigger.go:589]: this const is used also in `baseProcessor.go` file

**Review by iulianpascalau — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

File: node/chainSimulator/chainSimulator.go [hunk 3]
```diff
@@ -163,6 +163,7 @@ func (s *simulator) incrementRoundOnAllValidators() {
 
 func (s *simulator) allNodesCreateBlocks() error {
 	for _, node := range s.handlers {
+		time.Sleep(2 * time.Millisecond)
```
Comment (iulianpascalau) [inline node/chainSimulator/chainSimulator.go:166]: you will have a conflict here with what we have on staking-v4 branch:
```
// TODO MX-15150 remove this when we remove all goroutines
time.Sleep(2 * time.Millisecond)

err := node.CreateNewBlock()
if err != nil {
	return err
}
```

**Review by iulianpascalau — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
Normal allin test: v1.6.13-dev-config-06f147d698 -> move-in-config.toml-epoch--9b5f0ab6cf

--- Specific errors ---

block hash does not match 912
wrong nonce in block 301
miniblocks does not match 0
num miniblocks does not match 0
miniblock hash does not match 0
block bodies does not match 0
receipts hash missmatch 0

/------/


--- Statistics ---

Nr. of all ERRORS:  0
Nr. of all WARNS:  216
Nr. of new ERRORS:  0
Nr. of new WARNS:  0
Nr. of PANICS:  0

/------/

--- ERRORS ---

/------/

--- WARNINGS ---

/------/

**Pull request merged**